File: 605._Can_Place_Flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed, n):
        strr= ''.join(map(str,flowerbed))
        print(strr)
        res  = strr.replace('010', 'xxx').replace('10', 'xx').replace('01', 'xx').replace('00', '0x')
        print(res)
        return res.count('0') >= n 
        

File: 605._Can_Place_Flowers.cpp
bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int end = flowerbed.size()-1;
        for(int i = 0,count = 1;i <= end && n > 0;i++){
            count = flowerbed[i]?0:count+1;
            if(3 == count || (2 == count && i == end)){
                n--;
                count = 1;
            }
        }
        return 0 == n;
    }


